rct229/rules/section22/section22rule10.py

```diff
@@ -0,0 +1,148 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.baseline_systems.baseline_system_util import HVAC_SYS
+from rct229.ruleset_functions.get_baseline_system_types import get_baseline_system_types
+from rct229.ruleset_functions.get_primary_secondary_loops_dict import (
+    get_primary_secondary_loops_dict,
+)
+from rct229.schema.config import ureg
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.pint_utils import ZERO
+
+APPLICABLE_SYS_TYPES = [
+    HVAC_SYS.SYS_7,
+    HVAC_SYS.SYS_8,
+    HVAC_SYS.SYS_11_1,
+    HVAC_SYS.SYS_11_2,
+    HVAC_SYS.SYS_12,
+    HVAC_SYS.SYS_13,
+    HVAC_SYS.SYS_7B,
+    HVAC_SYS.SYS_8B,
+    HVAC_SYS.SYS_11_1B,
+    HVAC_SYS.SYS_12B,
+]
+PUMP_SPEED_CONTROL = schema_enums["PumpSpeedControlOptions"]
+MAX_FIXED_SPEED_CHW_LOOP_COOLING_CAPACITY = 300.0 * ureg("ton")
+
+
+class Section22Rule10(RuleDefinitionListIndexedBase):
+    """Rule 10 of ASHRAE 90.1-2019 Appendix G Section 22 (Chilled water loop)"""
+
+    def __init__(self):
+        super(Section22Rule10, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section22Rule10.PrimaryFluidLoopRule(),
+            index_rmr="baseline",
+            id="22-10",
+            description="For Baseline chilled water system with cooling capacity less than 300ton, the secondary pump shall be modeled as riding the pump curve. For Baseline chilled water system with cooling capacity of 300 tons or more, the secondary pump shall be modeled with variable-speed drives.",
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.fluid_loops[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        baseline_system_types_dict = get_baseline_system_types(rmi_b)
+        # create a list containing all HVAC systems that are modeled in the rmi_b
+        available_type_list = [
+            hvac_type
+            for hvac_type in baseline_system_types_dict.keys()
+            if len(baseline_system_types_dict[hvac_type]) > 0
+        ]
+
+        primary_secondary_loop_dict = get_primary_secondary_loops_dict(rmi_b)
+
+        return (
+            any(
+                [
+                    available_type in APPLICABLE_SYS_TYPES
+                    for available_type in available_type_list
+                ]
+            )
+            and len(primary_secondary_loop_dict) > 0
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+
+        loop_pump_dictionary = {
+            pump["loop_or_piping"]: pump for pump in find_all("$.pumps[*]", rmi_b)
```

Comment:
Nice!

Reply:
We may want to use dict<string: list[pump]> as the data type (align with RDS) for `loop_pump_dictionary` because multiple pumps in a loop is possible.

Reply:
Fixed. I'd appreciate it if you could check this part one more time.